### PR TITLE
Script for defect 2481

### PR DESCRIPTION
**Pull request by TMyza**
ATF Test Scripts to check [#2481](https://github.com/smartdevicelink/sdl_core/issues/2481)

Summary
A set of test scripts to verify that issue is fixed

This PR is **not ready** for review.
This is **draft** version of the script.

### CODE REVIEW

File: test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+-- 1) Vehicle Ignition On and Running
+-- 2) Android connected via BT
+-- 3) SyncProxyTester Media App running on phone and SYNC
+-- 4) Created Interaction Choice Set with ID 1 and data items as A, B and C respectively
+-- Steps to reproduce:
+-- 1) Select Send Message -> PerformInteraction and enter following data:
+--    Set the Initial text as 'Pick an Alphabet:', Initial Prompt as 'Pick a command', Interaction Mode as 'BOTH',
+--    Choice Set ID as '2', Check help prompt, timeout prompt and set timeout as 10000. Uncheck the VR help Item #. Press Ok.  
+-- Expected:
+-- 1) PerformInteraction(request) notificaiton is displayed on the app.
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua]: typo 'notification'

File: test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  local tmp = {
+    {
+      text = text,
+      type = "TEXT"
+    }
+  }
+  return tmp
+end
+
+local initialPromptValue = PromptValue(" Make your choice ")
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua]: is not used

File: test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+local helpPromptValue = PromptValue(" Help Prompt ")
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua]: is not used

File: test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+local timeoutPromptValue = PromptValue(" Time out ")
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua]: is not used

File: test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+local paramsBoth = {
+    initialText = "Pick an Alphabet",
+    initalPrompt = "Pick a command",
+    interactionMode = "BOTH",
+    interactionChoiceSetIDList = { 1 }
+}
+
+-- [[ Local Functions ]]
+local function setExChoiceSet(choiceIDValues)
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua]: interactionChoiceSetIDList contains only one value, so params can be defined without  additional function.

File: test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    :Do(function(_, data)
+        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+    common.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end 
+
+local function sendPerformInteraction(pParam)
+    local cid = common.getMobileSession():SendRPC("PerformInteraction", pParam)
+    common.getHMIConnection():ExpectRequest("VR.PerformInteraction", {
+      helpPrompt = pParam.helpPrompt,
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua]: values for prompt must be like propmtValue

File: test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+      initialPrompt = pParam.initialPrompt,
+      timeout = pParam.timeout,
+      timeoutPrompt = pParam.timeoutPrompt      
+    })
+    :Do(function(_,data)
+      common.getHMIConnection():SendNotification("VR.Started")
+      common.getHMIConnection():SendNotification("TTS.Started")
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua]: i see 3 responses to VR.PI .first

File: test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+      SendOnSystemContext("VRSESSION")
+      local function firstSpeakTimeOut()
+        common.getHMIConnection():SendNotification("VR.Stopped")
+        common.getHMIConnection():SendNotification("TTS.Stopped")
+        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua]: second one

File: test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+      end
+      RUN_AFTER(firstSpeakTimeOut, 5)
+      local function vrResponse()
+        common.getHMIConnection():SendError(data.id, data.method, "TIMED_OUT", "Perform Interaction error response.")
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua]: third one. Also according to defect description choiceId must be send from VR.PI

File: test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+        fieldText = pParam.initialText
+      },
+      vrHelp = pParam.vrHelp,
+      vrHelpTitle = pParam.initialText
+    })
+  :Do(function(_,data)
+    SendOnSystemContext("HMI_OBSCURED")
+    local function uiResponse()
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS",
+        { choiceID = pParam.interactionChoiceSetIDList[1] })
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua]: UI.PI response must be without choiceID

File: test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua [hunk 11]
```diff
@@ -0,0 +1,144 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/smartdevicelink/sdl_core/issues/2481
+-- Description:
+-- 1) ChoiceID is missing in PerformInteraction response
+-- Precondition:
```
Comment (aderiabin) [inline test_scripts/Defects/5_0/2481_ChoiceID_is_missing_in_PerformInteraction_response.lua]: @TMyza Please modify precondition and steps to reproduce for them become applicable to automated script.

**Comment by dboltovskyi** [comment]
Livio team please do not review this PR for now since it hasn't passed our internal review yet.
Review and appropriate updates going be done till 1st of November.
Thank you.

**Comment by dboltovskyi** [comment]
The internal review iteration has been moved to the LuxoftSDL.

**Pull request closed without merging**